src/components/batch-selection/batch-selection.pw.tsx

```diff
@@ -0,0 +1,191 @@
+import React from "react";
+import { test, expect } from "@playwright/experimental-ct-react17";
+import { BatchSelectionProps } from "components/batch-selection";
+import BatchSelectionComponent from "./components.test-pw";
+import {
+  checkAccessibility,
+  positionOfElement,
+  checkGoldenOutline,
+} from "../../../playwright/support/helper";
+
+import {
+  batchSelectionCounter,
+  batchSelectionComponent,
+  batchSelectionButtonsByPosition,
+} from "../../../playwright/components/batch-selection";
+
+const BATCH_SELECTION_COLOR = [
+  "dark",
+  "light",
+  "white",
+  "transparent",
+] as const;
+
+test.describe("check BatchSelection component properties", () => {
+  [0, 10, 100].forEach((selectedCount) => {
+    test(`check BatchSelection component ${selectedCount} selected Count`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<BatchSelectionComponent selectedCount={selectedCount} />);
+
+      await expect(batchSelectionCounter(page)).toHaveText(
+        `${selectedCount} selected`
+      );
+    });
+  });
+
+  test("should check hidden BatchSelection", async ({ mount, page }) => {
+    await mount(<BatchSelectionComponent hidden />);
+
+    await expect(batchSelectionComponent(page).getAttribute("hidden"))
+      .toBeTruthy;
+
+    await expect(batchSelectionComponent(page)).toBeVisible();
```

Comment:
**issue:** I wouldn't expect `.toBeVisible()`to pass here considering that the prop is called `hidden`. I'd expect the correct assertion to be `.not.toBeVisible()`.